### PR TITLE
Icon colors based on tracker name

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203581260317501/f

### Description
When we don't have the icon of the tracking company we create an icon on
the fly with the first letter but fixed background color.

This PR changes the logic so that the color is picked from a color
pallete and calculated directly from the company name text

### Steps to test this PR
Smoke tests for AppTP

_Icon colors_
- [x] install from this branch
- [x] enable AppTP
- [x] install termux app
- [x] execute `curl -v <domain> &` for the following domains, ie. `https://logs.datadoghq.com` and `https://mobile.adsafeprotected.com`
- [x] go to AppTP activity screen
- [x] verify icons have colors
- [x] go to AppTP detail screen (tap on the termux entry)
- [x] verify icons have the same color
- [x] close and relaunch the app
- [x] verify the colors continue to be the same in above screens

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3034** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3034" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3034?utm_source=stack-comment).

**Comment by nalcalag** [comment]
Maybe I'm wrong, but as far as I can see the custom view `TextDrawable` is not a component defined in the design system and it's only used in AppTP. Therefore, I would suggest to move it to `vpn-impl` module.
The main reason it's that we are adding a dependency only because this feature which will be added to all modules we are using `common-ui`.

**Comment by aitorvs** [comment]
> Maybe I'm wrong, but as far as I can see the custom view `TextDrawable` is not a component defined in the design system and it's only used in AppTP. Therefore, I would suggest to move it to `vpn-impl` module. The main reason it's that we are adding a dependency only because this feature which will be added to all modules we are using `common-ui`.

I can move it yup. Didn't touch it because it was originally in there.
FWIW okio is already added as a dependency in the app (directly and indirectly), so moving or not moving doesn't change a thing on that front.

File: common-ui/build.gradle [hunk 1]
```diff
@@ -46,6 +46,7 @@ dependencies {
     implementation AndroidX.constraintLayout
     implementation AndroidX.recyclerView
     implementation AndroidX.lifecycle.viewModelKtx
+    implementation Square.okio
```
Comment (nalcalag) [inline common-ui/build.gradle:49]:
```suggestion
```
We don't need this anymore 👍

**Review by karlenDimla — APPROVED** [review]
Tested and logic make sense.

**Review by nalcalag — APPROVED** [review]
Thanks for moving the custom view `TextDrawable` to `vpn-impl` method 🙏

**Pull request merged**